crates/compiler/builtins/bitcode/src/dec.zig

```diff
@@ -682,16 +685,34 @@ fn div_u256_by_u128(numer: U256, denom: u128) U256 {
         lo_overflowed = @subWithOverflow(u128, lo, 1, &lo);
         hi = hi -% @intCast(u128, @bitCast(u1, lo_overflowed));
 
-        // TODO this U256 was originally created by:
+        // NOTE: this U256 was originally created by:
         //
         // ((hi as i128) >> 127).as_u256()
         //
-        // ...however, I can't figure out where that function is defined.
-        // Maybe it's defined using a macro or something. Anyway, hopefully
-        // this is what it would do in this scenario.
+        // As an implementation of `as_u256`, we wrap a negative value around to the maximum value of U256.
+
+        var hi_i128: i128 = undefined;
+        if (hi > i128_max) {
```

Comment:
Shouldn't this be the same behavior as doing `var hi_i128 = @bitCast(i128, hi)`? If I understand correctly, adding the overflow the the min negative value I believe is the same as bit cast from a `u128` to an `i128`

Reply:
You are right. `@bitCast` was exactly what I wanted to do.
This part is gone tho.